controllers/webhook/ksvc_validator.go

```diff
@@ -0,0 +1,118 @@
+package webhook
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	kservev1beta1 "github.com/kserve/kserve/pkg/apis/serving/v1beta1"
+	"k8s.io/apimachinery/pkg/runtime"
+	types2 "k8s.io/apimachinery/pkg/types"
+	knservingv1 "knative.dev/serving/pkg/apis/serving/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+
+	"github.com/opendatahub-io/odh-model-controller/controllers/constants"
+	"github.com/opendatahub-io/odh-model-controller/controllers/resources"
+)
+
+// +kubebuilder:webhook:admissionReviewVersions=v1,path=/validate-serving-knative-dev-v1-service,mutating=false,failurePolicy=fail,groups="serving.knative.dev",resources=services,verbs=create,versions=v1,name=validating.ksvc.odh-model-controller.opendatahub.io,sideEffects=None
+
+type ksvcValidator struct {
+	client client.Client
+}
+
+func (v *ksvcValidator) ValidateCreate(ctx context.Context, obj runtime.Object) error {
+	log := logf.FromContext(ctx).WithName("KsvcValidatingWebhook")
```

Comment:
@bartoszmajsak @retocode @skonto May I have a review from you?
It doesn't need to be a deep review. I simply want to know if this approach makes sense to you.

Reply:
I can't speak for the istio side of things (maybe there is another way to enforce this?), but from OSS perspective, having a validating webhook makes sense to me.